.github/workflows/template-version.yml

```
@@ -0,0 +1,49 @@
+        run: |
+          if git diff --name-only ${{ github.base_ref }} ${{ github.head_ref }} | grep -q '^templates/autograded/'; then
+            echo "autograded_changed=true" >> $GITHUB_ENV
+          else
+            echo "autograded_changed=false" >> $GITHUB_ENV
+          fi
+      - name: Update autograded template version
+        if: env.autograded_changed == 'true'
+        run: |
+          sed -i "s/\"version\": \".*\"/\"version\": \"$(date +"%Y.%m.%d-%H%M")\"/" templates/autograded/version.json
+      - name: Check if manual files were changed
+        run: |
```

actionlint (1.7.4)

"github.head_ref" is potentially untrusted. avoid using it directly in inline scripts. instead, pass it through an environment variable. see https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions for more details

(expression)